modules/generator/storage/instance.go

```diff
@@ -60,14 +60,22 @@ var _ Storage = (*storageImpl)(nil)
 // Should we standarize slog and deprecate go-kit/log too?
 func New(cfg *Config, o Overrides, tenant string, reg prometheus.Registerer, _ log.Logger) (Storage, error) {
 	// TODO move this to the generator.go
+
+	// Validate empty tenant to prevent WAL directory deletion
+	if tenant == "" {
```

Comment:
i like this fix b/c of how catastrophic the result is if it occurs, but technically this scenario should never happen.

wdyt about just returning an error instead of using "single-tenant"?

Reply:
@joe-elliott 
I agree thx

Finally, I used `RemoteWriteAddOrgIDHeader` as the decision flag because it already maps directly to the deployment mode:

- true (multi-tenant) → tenant ID is required for isolation, so we throw an error if it’s missing
- false (single-tenant) → no isolation needed, so we allow empty tenant IDs to pass through

This way we don’t introduce new config knobs or break existing single-tenant flows. 
The validation logic stays aligned with the original flag’s semantics, making the behavior both clear and consistent.